src/main/java/com/eppo/sdk/dto/BanditLogData.java

```diff
@@ -1,39 +1,47 @@
 package com.eppo.sdk.dto;
 
 import java.util.Date;
+import java.util.Map;
 
 /**
  * Assignment Log Data Class
  */
 public class BanditLogData {
-    public String experiment;
-    public String variation;
-    public String subject;
-    public EppoAttributes subjectAttributes;
-    public String action;
-    public EppoAttributes actionAttributes;
-    public Float actionProbability;
-    public String modelVersion;
-    public Date timestamp;
+  public Date timestamp;
 
-    public BanditLogData(
-            String experiment,
-            String variation,
-            String subject,
-            EppoAttributes subjectAttributes,
-            String action,
-            EppoAttributes actionAttributes,
-            Float actionProbability,
-            String modelVersion
-    ) {
-        this.experiment = experiment;
-        this.variation = variation;
-        this.subject = subject;
-        this.subjectAttributes = subjectAttributes;
-        this.action = action;
-        this.actionAttributes = actionAttributes;
-        this.actionProbability = actionProbability;
-        this.modelVersion = modelVersion;
-        this.timestamp = new Date();
-    }
+  public String experiment;
+  public String variation;
+  public String subject;
+  public String action;
+  public Float actionProbability;
+  public String modelVersion;
+  public Map<String, Double> subjectNumericAttributes;
+  public Map<String, String> subjectCategoricalAttributes;
+  public Map<String, Double> actionNumericAttributes;
+  public Map<String, String> actionCategoricalAttributes;
+
+  public BanditLogData(
+    String experiment,
+    String variation,
+    String subject,
+    String action,
+    Float actionProbability,
```

Comment:
Maybe this should be `Double`, since `computeActionWeights` returns probabilities/weights as doubles.

Reply:
Oh yes great catch!